File: java/src/main/java/com/android/frame/guide/SimpleViewAdapter.java
package com.android.frame.guide;

import android.view.View;
import android.view.ViewGroup;

import androidx.viewpager.widget.PagerAdapter;

import java.util.List;

/**
 * Created by xuzhb on 2020/7/4
 * Desc:
 */
public class SimpleViewAdapter extends PagerAdapter {

    private List<View> viewList;  //数据源

    public SimpleViewAdapter(List<View> viewList) {
        this.viewList = viewList;
    }

    //数据源的数目
    public int getCount() {
        return viewList.size();
    }

    //view是否由对象产生，官方写arg0==arg1即可
    public boolean isViewFromObject(View arg0, Object arg1) {
        return arg0 == arg1;
    }

    //销毁一个页卡(即ViewPager的一个item)
    public void destroyItem(ViewGroup container, int position, Object object) {
        container.removeView(viewList.get(position));
    }

    //对应页卡添加上数据
    public Object instantiateItem(ViewGroup container, int position) {
        container.addView(viewList.get(position));//千万别忘记添加到container
        return viewList.get(position);
    }

}

File: kotlin/src/main/java/com/android/frame/guide/SimpleViewAdapter.kt
package com.android.frame.guide

import android.view.View
import android.view.ViewGroup
import androidx.viewpager.widget.PagerAdapter

/**
 * Created by xuzhb on 2020/12/28
 * Desc:
 */
class SimpleViewAdapter(private val viewList: MutableList<View>) : PagerAdapter() {

    //view是否由对象产生，官方写view == `object`即可
    override fun isViewFromObject(view: View, `object`: Any): Boolean = view == `object`

    //数据源的数目
    override fun getCount(): Int = viewList.size

    //销毁一个页卡(即ViewPager的一个item)
    override fun destroyItem(container: ViewGroup, position: Int, `object`: Any) {
        container.removeView(viewList[position])
    }

    //对应页卡添加上数据
    override fun instantiateItem(container: ViewGroup, position: Int): Any {
        container.addView(viewList[position])  //千万别忘记添加到container
        return viewList[position]
    }

}